## Action goal

# Number of objects to pick-place
int32 n_objects
# Goal for sending waypoint path to follow
ir2425_group_15/RobotNavigationGoal robot_navigation_goal
# Goal for sending AprilTag IDs of objects to pick-place
ir2425_group_15/AprilTagsDetectionGoal apriltags_detection_goal
# Goal for sending straight line coefficients
ir2425_group_15/PickPlaceGoal pick_place_goal

---

## Action result

# Task status
string task_state
# Complete flag
bool is_completed
# Results from robot navigation, AprilTags detection
ir2425_group_15/RobotNavigationResult robot_navigation_result
ir2425_group_15/AprilTagsDetectionResult apriltags_detection_result

---

## Action feedback

# Verbose flag
bool verbose
# Task status
string task_state
# Feedbacks from robot navigation, robot controller, AprilTags detection, pick-place
ir2425_group_15/RobotNavigationFeedback robot_navigation_feedback
ir2425_group_15/AprilTagsDetectionFeedback apriltags_detection_feedback
ir2425_group_15/RobotControllerFeedback robot_controller_feedback
ir2425_group_15/PickPlaceResult pick_place_feedback